internal/monitors/ecs/ecs.go

```diff
@@ -0,0 +1,275 @@
+	// label values will become the dimension value for the mapped name.  E.g.
+	// `io.kubernetes.container.name: container_spec_name` would result in a
+	// dimension called `container_spec_name` that has the value of the
+	// `io.kubernetes.container.name` container label.
+	LabelsToDimensions map[string]string `yaml:"labelsToDimensions"`
+	// A list of filters of images to exclude.  Supports literals, globs, and
+	// regex.
+	ExcludedImages []string `yaml:"excludedImages"`
+}
+
+// Monitor for ECS Metadata
+type Monitor struct {
+	Output  types.Output
+	cancel  func()
+	client  *http.Client
+	conf *Config
+	ctx context.Context
+	timeout time.Duration
+	taskDimensions map[string]string
+	containers map[string]ecs.Container
+	// shouldIgnore - key : container docker id, tells if stats for the container should be ignored.
+	// Usually the container was filtered out by excludedImages
+	// or container metadata is not received.
+	shouldIgnore map[string]bool
+	imageFilter filter.StringFilter
+}
+
+// Configure the monitor and kick off volume metric syncing
+func (m *Monitor) Configure(conf *Config) error {
+	var err error
+	m.imageFilter, err = filter.NewExhaustiveStringFilter(conf.ExcludedImages)
+	if err != nil {
+		return errors.Wrapf(err, "Could not load excluded image filter")
+	}
+
+	m.conf = conf
+	m.timeout = time.Duration(conf.TimeoutSeconds) * time.Second
+	m.client = &http.Client{
+		Timeout: m.timeout,
+	}
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+
+	isRegistered := false
+
+	utils.RunOnInterval(m.ctx, func() {
+		if !isRegistered {
+			task, err := fetchTaskMetadata(m.client, m.conf.MetadataEndpoint)
+			if err != nil {
+				logger.WithFields(log.Fields{
+					"error":    err,
+				}).Error("Could not receive ECS Task Metadata")
+				m.cancel()
+			}
+			m.taskDimensions = task.GetDimensions()
+			m.containers, m.shouldIgnore = parseContainers(task, m.imageFilter)
+			
+			isRegistered = true
+		}
+
+		m.fetchStatsForAll()
+	}, time.Duration(conf.IntervalSeconds)*time.Second)
+
+	return nil
+}
+
+// Fetch a container with given container docker ID and load it to the monitor
+// If the container is successfully received, return true. Else, return false
+func (m *Monitor) fetchContainer(dockerId string) (ecs.Container, error) {
+	body, err := getMetadata(m.client, getURI(m.conf.MetadataEndpoint, dockerId))
+	if err != nil {
+		return ecs.Container{}, err
+	}
+
+	var container ecs.Container
+
+	if err := json.Unmarshal(body, &container); err != nil {
+		return ecs.Container{}, errors.Wrapf(err, "Could not parse ecs container json")
+	}
+
+	if (m.imageFilter != nil && m.imageFilter.Matches(container.Image)) ||
+		container.Type != "NORMAL" {
+		return ecs.Container{}, errors.New("The container has been excluded by image filter.")
+	}
+
+	return container, nil
+}
+
+func (m *Monitor) fetchStatsForAll() {
+	_, cancel := context.WithTimeout(m.ctx, m.timeout)
```

Comment:
If you don't use the context returned, there is no point in calling the `cancel` method, as it only acts on the context you are ignoring with `_` here.

Reply:
You probably don't need any context here.